CDTDatastore/query/CDTQIndex.m

```diff
@@ -82,6 +86,23 @@ + (instancetype)index:(NSString *)indexName
                ofType:(NSString *)indexType
          withSettings:(NSDictionary *)indexSettings
 {
+    CDTQIndexType typeAsEnum = [CDTQIndexManager indexTypeForString:indexType];
+    return [[self class] index:indexName withFields:fieldNames type:typeAsEnum withSettings:indexSettings];
+}
+
+#pragma mark New creator methods
+
+
++ (instancetype)index:(NSString *)indexName withFields:(NSArray<NSString *> *)fields type:(CDTQIndexType)type
+{
+        return [[self class] index:indexName withFields:fields type:type withSettings:nil];
+}
+
++ (instancetype)index:(NSString *)indexName
+           withFields:(NSArray *)fieldNames
+                 type:(CDTQIndexType)indexType
+         withSettings:(NSDictionary *)indexSettings
```

Comment:
We are a bit hamstrung, I guess we could go with `withType` for `CDTQIndex` and `ofType` where there is `withType`?